src/main/java/emu/grasscutter/scripts/ScriptLib.java

```diff
@@ -1013,22 +1013,38 @@ public int ChangeToTargetLevelTag(int var1){
     }
 
     public int AddSceneTag(int sceneId, int sceneTagId){
-        logger.warn("[LUA] Call unimplemented AddSceneTag with {}, {}", sceneId, sceneTagId);
-        //TODO implement
+        logger.debug("[LUA] Call AddSceneTag with {}, {}", sceneId, sceneTagId);
+        // Ensure key exists for given scene
+        if (sceneScriptManager.get().getScene().getHost().getSceneTags().get(sceneId) == null) {
+            sceneScriptManager.get().getScene().getHost().getSceneTags().put(sceneId, new HashSet<>());
+        }
+        sceneScriptManager.get().getScene().getHost().getSceneTags().get(sceneId).add(sceneTagId);
         return 0;
     }
 
     public int DelSceneTag(int sceneId, int sceneTagId){
-        logger.warn("[LUA] Call unimplemented DelSceneTag with {}, {}", sceneId, sceneTagId);
-        //TODO implement
+        logger.debug("[LUA] Call DelSceneTag with {}, {}", sceneId, sceneTagId);
+        var sceneTags = sceneScriptManager.get().getScene().getHost().getSceneTags();
+        // Ensure key exists for given scene
+        if (sceneTags.get(sceneId) == null) {
+            // Can't delete something that doesn't exist
+            return 0;
+        }
+        sceneScriptManager.get().getScene().getHost().getSceneTags().get(sceneId).remove(sceneTagId);
```

Comment:
There are a lot of things managing scenetags, so it might be good to move the general add/remove logic to the player progress Manager.
Im also relatively sure, we should send some type of package when a tag gets added/removed, to notify the client about it.